meta-lmp-bsp/recipes-security/optee/optee-os-fio_3.17.0.bbappend

```diff
@@ -1 +1,20 @@
 require optee-os-fio-bsp.inc
+
+OPTEE_SUFFIX:stm32mp1common    = "${@bb.utils.contains('MACHINE_FEATURES','fip','bin','stm32',d)}"
+OPTEE_HEADER:stm32mp1common    = "tee-header_v2"
+OPTEE_PAGEABLE:stm32mp1common  = "tee-pageable_v2"
+OPTEE_PAGER:stm32mp1common     = "tee-pager_v2"
```

Comment:
In theory these files are already deployed by default at the deploy / machine / optee folder.

Reply:
fip bbclacs  from `meta-st-stm32mp1` expects this naming format of optee images, so this is why I added that additional `do_deploy()` step:

in `meta-st-stm32mp1/classes/fip-utils-stm32mp.bbclass`:
```
206     elif [ "${bl32_conf}" = "optee" ]; then                             
207                 # Check for files                                               
208                 [ -f "${FIP_DEPLOYDIR_OPTEE}/${FIP_OPTEE_HEADER}-${dt}.${FIP_OPTEE_SUFFIX}" ] || bbfatal "Missing ${FIP_OPTEE_HEADER}-${dt}.${FIP_OPTEE_SUFFIX} file in folder: ${FIP_DEPLOYDIR_OPTEE}"
209                 [ -f "${FIP_DEPLOYDIR_OPTEE}/${FIP_OPTEE_PAGER}-${dt}.${FIP_OPTEE_SUFFIX}" ] || bbfatal "Missing ${FIP_OPTEE_PAGER}-${dt}.${FIP_OPTEE_SUFFIX} file in folder: ${FIP_DEPLOYDIR_OPTEE}"
210                 [ -f "${FIP_DEPLOYDIR_OPTEE}/${FIP_OPTEE_PAGEABLE}-${dt}.${FIP_OPTEE_SUFFIX}" ] || bbfatal "Missing ${FIP_OPTEE_PAGEABLE}-${dt}.${FIP_OPTEE_SUFFIX} file in folder: ${FIP_DEPLOYDIR_OPTEE}"
211                 # Set FIP_EXTRACONF                                             
212                 FIP_EXTRACONF="\                                                
213                     --tos-fw ${FIP_DEPLOYDIR_OPTEE}/${FIP_OPTEE_HEADER}-${dt}.${FIP_OPTEE_SUFFIX} \
214                     --tos-fw-extra1 ${FIP_DEPLOYDIR_OPTEE}/${FIP_OPTEE_PAGER}-${dt}.${FIP_OPTEE_SUFFIX} \
215                     --tos-fw-extra2 ${FIP_DEPLOYDIR_OPTEE}/${FIP_OPTEE_PAGEABLE}-${dt}.${FIP_OPTEE_SUFFIX} \
216                     "                                                           

```

Reply:
Sure, but then we just need the if block, not else, as those files are already created and deployed by default (with similar name).